cmd/libsnap-confine-private/device-cgroup-support.c

```diff
@@ -351,10 +351,13 @@ static int _sc_cgroup_v2_init_bpf(sc_device_cgroup *self, int flags) {
     /* copy what fits into BPF_OBJ_NAME_LEN-2 ('s_' prefix) and truncate the
        rest */
     strncpy(self->v2.name + 2, snap_name_start, sizeof(self->v2.name) - 2 - 1);
-    /* replace all hyphens which are valid in security tags, but are invalid for
-     * BPF object names */
-    for (char *c = strchr(self->v2.name, '-'); c != NULL; c = strchr(c, '-')) {
+    /* Security tags may contain '+' and '-' characters that are not
```

Comment:
component hooks?

Reply:
this was discussed later again, the name in question here is only effectively used for presentation, i.e. the thing you'd see in `bpftool prog list` or `bpftool map list`.